mlir/include/mlir/Dialect/Vector/IR/VectorOps.td

```diff
@@ -789,6 +789,57 @@ def Vector_FMAOp :
   }];
 }
 
+def Vector_ToElementsOp : Vector_Op<"to_elements", [
+    Pure,
+    TypesMatchWith<"operand element type matches result types",
+                   "input", "elements", "SmallVector<Type>("
+                   "::llvm::cast<VectorType>($_self).getNumElements(), "
+                   "::llvm::cast<VectorType>($_self).getElementType())">]> {
```

Comment:
Could this be shared with `vector.from_elements`?

Reply:
The check is inverse so I'm not sure how to create a record that doesn't require to provide the same information here